SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDevAuthTokenGenerator.cpp

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialGDKDevAuthTokenGenerator.h"
+
+#include "Async/Async.h"
+#include "Framework/Notifications/NotificationManager.h"
+
+#include "SpatialCommandUtils.h"
+#include "SpatialGDKEditorSettings.h"
+#include "SpatialGDKSettings.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialGDKDevAuthTokenGenerator);
+
+FSpatialGDKDevAuthTokenGenerator::FSpatialGDKDevAuthTokenGenerator()
+	: bIsGenerating(false)
+{
+}
+
+void FSpatialGDKDevAuthTokenGenerator::UpdateSettings(FString DevAuthToken)
+{
+	USpatialGDKEditorSettings* GDKEditorSettings = GetMutableDefault<USpatialGDKEditorSettings>();
+	GDKEditorSettings->DevelopmentAuthenticationToken = DevAuthToken;
+	GDKEditorSettings->SaveConfig();
+	GDKEditorSettings->SetRuntimeDevelopmentAuthenticationToken();
+	GDKEditorSettings->SetRuntimeUseDevelopmentAuthenticationFlow();
+}
+
+void FSpatialGDKDevAuthTokenGenerator::DoGenerateDevAuthTokenTasks()
+{
+	bool bIsRunningInChina = GetDefault<USpatialGDKSettings>()->IsRunningInChina();
+	AsyncTask(ENamedThreads::AnyBackgroundThreadNormalTask, [this, bIsRunningInChina]
+	{
+		AsyncTask(ENamedThreads::GameThread, [this]()
+		{
+			this->ShowTaskStartedNotification(TEXT("Generating Development Authentication Token"));
+		});
+
+		FString DevAuthToken;
+		FString ErrorMessage;
+		if (SpatialCommandUtils::GenerateDevAuthToken(bIsRunningInChina, DevAuthToken, ErrorMessage))
```

Comment:
:o 

Thank you for using SpatialCommandUtils <3

Reply:
Any timeout on this? If this never returns for whatever reason the notification will never stop?